Kickstarter-iOS/AppDelegate.swift

```diff
@@ -263,6 +263,8 @@ internal final class AppDelegate: UIResponder, UIApplicationDelegate {
 
     UNUserNotificationCenter.current().delegate = self
 
+    UIApplication.shared.registerForRemoteNotifications()
```

Comment:
It looks like we have a bunch of stuff in AppDelegateViewModel and PushRegistration already which this seems to bypass, can we use that object for this and update any signals if they aren't firing, rather than call this manually here?

Reply:
The `PushRegistration.register` method is only triggered once the user interacts with the request permissions dialog. 

I can add a `registerOnAppLaunch` method to `PushRegistration` and set up a new input that then allows us to call `UIApplication.shared.registerForRemoteNotifications()` from the view model. This initially felt a little over-engineered to me, though.

Reply:
Here's a [solution](https://github.com/kickstarter/ios-oss/pull/2172/commits/e10a7e890a773e8ee3b31637dce8831970386aff) that uses the existing PushRegistration class. Lmk what ya think!

Reply:
- In cases where the user has already authorized notifications, does the `UIApplication.shared.registerForRemoteNotifications()` call happen in PushRegistration?
- In cases where the user has not authorized, does this trigger the permission dialog? I suspect without the call to `UNUserNotificationCenter.current().requestAuthorization` that it might fail. But either way we should control when the dialog appears.

I don't think we need a new method for PushRegistration but we can just stick something into app launch signals to trigger the existing code path.

Reply:
Yeah the existing code path registers for remote notifications, but only after requesting authorization, and we present the request alert once the user has logged in and opted in to receive notifications via a custom KS alert that is shown just before the official Apple-generated permissions dialog.

The commit I just put up makes the call to register via the initial app launch signal like we want it to. I like this because it keeps the logic that requests PN authorization separate from registering.